src/main/java/com/dataartisans/training/TroubledStreamingJob.java

```diff
@@ -42,18 +52,26 @@ public static void main(String[] args) throws Exception {
         //Checkpointing Configuration
         env.enableCheckpointing(5000);
         env.getCheckpointConfig().setMinPauseBetweenCheckpoints(4000);
+
         if (env instanceof LocalStreamEnvironment) {
-            // configure checkpoint storage for local executions
-            String statePath = parameters.get("fsStatePath", "file:///" + System.getProperty("user.dir") + "/chkpts");
-            FileUtils.deleteDirectory(new File(new URI(statePath))); // cleanup last executions
-            StateBackend stateBackend = new FsStateBackend(statePath);
+            String statePath = parameters.get("fsStatePath");
```

Comment:
* Did you verify that this works on Windows as well?
* Let's pull this up into the first `local`-conditional.

Reply:
No, I do not have a windows machine to test against. I was just going with the javadoc for createTempDirectory and Path.fromLocalFile that claim windows compatibility. And if an explicit fsStatePath is provided then it is the users responsibility to provide a valid path for their os.